tests/ci_install.sh

```diff
@@ -34,26 +33,24 @@ if [[ "${RUNNER_OS}" == "Windows" ]] && [[ ${dependencies} != "minimal" ]]; then
     while read requirement; do conda install --yes $requirement; done < tests/windows_conda_requirements.txt
 else
     python -m pip install --upgrade pip
```

Comment:
I should also add a comment here why we upgrade pip (it's needed in most Python versions to support `pip install .[full]`, see #4270)